File: Wirtualny_Swiat_Java/Projekt_PO_Java/src/model/Organisms/Berries.java
package model.Organisms;

import model.Organism;
import model.Plant;
import model.Point;
import model.World;

import java.awt.*;

public class Berries extends Plant {
    private final Color berriesColor = Color.MAGENTA;

    public Berries(World world, Point position){
        super(world, position, 99);
    }

    @Override
    public String getNazwa(){
        return "WilczeJagody";
    }

    @Override
    public Color getColor(){
        return berriesColor;
    }

    @Override
    public Organism makeNewOrganism(Point position){
        return new Berries(world, position);
    }

    @Override
    public void Collision(Organism attacker){
        attacker.setAlive(false);
        world.addEvent(attacker.getNazwa() + " zjadł wilcze jaody i zmarl");
    }
}

File: WirtualnySwiatPython/Plants/Berries.py
from Animals.Human import Human
from plant import Plant
from config import PURPLE

class Berries(Plant):
    def __init__(self, world, position):
        super().__init__(world, position, power=99)
        self.color = PURPLE

    def getNazwa(self):
        return "WilczeJagody"

    def getColor(self):
        return self.color

    def makeNewOrganism(self, position):
        return Berries(self.world, position)

    def Collision(self, attacker):
        if isinstance(attacker, Human):
            self.world.killHuman()
            return

        attacker.setIsAlive(False)
        self.world.addEvent(f"{attacker.getNazwa()} zjadl wilcze jagody i zginal")